From a306bd79b87e8b8f396c30128358ca695e0ccd80 Mon Sep 17 00:00:00 2001
From: Klaus Ethgen <Klaus@Ethgen.de>
Date: Fri, 6 Dec 2019 18:22:09 +0100
Subject: [PATCH] Work around an missing property search

There might be PDF missing properties. But the root in the tree is not
undefined than PDF::API2::Basic::PDF::Null.

So the following find_prop will fail badly instead of just been ignored.
---
 lib/PDF/API2/Basic/PDF/Pages.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/PDF/API2/Basic/PDF/Pages.pm b/lib/PDF/API2/Basic/PDF/Pages.pm
index 87a4778..89aab26 100644
--- a/lib/PDF/API2/Basic/PDF/Pages.pm
+++ b/lib/PDF/API2/Basic/PDF/Pages.pm
@@ -267,7 +267,7 @@ sub find_prop {
             return $self->{$prop};
         }
     }
-    elsif (defined $self->{'Parent'}) {
+    elsif (defined $self->{'Parent'} and ref($self->{'Parent'}) ne 'PDF::API2::Basic::PDF::Null') {
         return $self->{'Parent'}->find_prop($prop);
     }
 
-- 
2.24.0

